zookeeper-server/src/main/java/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1284,10 +1284,11 @@ public void run() {
                 } catch (Throwable e) {
                     if (closing) {
                         // closing so this is expected
-                        LOG.warn(
-                            "An exception was thrown while closing send thread for session 0x{}.",
-                            Long.toHexString(getSessionId()),
-                            e);
+                        if (LOG.isDebugEnabled()) {
+                            LOG.debug(
+                                "An exception was thrown while closing send thread for session 0x{}.",
+                                Long.toHexString(getSessionId()), e);
+                        }
```

Comment:
IIUC `LOG.debug` already contains a call to `LOG.isDebugEnabled`.

Reply:
It is not a core code flow. So for me it is ok to go either way. 
I have put debug enable check to void unnecessary toHexString call.
shall we keep as it is or shall it change it?

Reply:
Thanks for your explanation. Then we can keep as is.